src/net.cpp

```diff
@@ -1924,16 +1922,28 @@ void CConnman::ThreadOpenConnections(const std::vector<std::string> connect)
         int nTries = 0;
         while (!interruptNet)
         {
-            CAddrInfo addr = addrman.SelectTriedCollision();
-
-            // SelectTriedCollision returns an invalid address if it is empty.
-            if (!fFeeler || !addr.IsValid()) {
-                addr = addrman.Select(fFeeler);
-            }
-
-            // Require outbound connections, other than feelers, to be to distinct network groups
-            if (!fFeeler && setConnected.count(addr.GetGroup(addrman.m_asmap))) {
-                break;
+            CAddrInfo addr;
+
+            // For non-feeler connections, attempt to connect to an address among the collisions
+            // in our "tried" table.
+            // For feelers iterations, since feelers should be selected from the "new" table in the AddrMan,
+            // skip selecting collisions.
+            // Also require non-feelers to be distinct network groups to preserve connection diversity.
+            // It's much less relevant for feelers, because it's a frequent procedure
+            // to sanitize AddrMan (not critical), and selecting from AddrMan is
+            // sufficiently (for this case) random/difficult-to-Sybil due to the bucketing.
+            if (conn_type != ConnectionType::FEELER) {
+                // SelectTriedCollision returns an invalid address if it is empty.
+                addr = addrman.SelectTriedCollision();
+                if (!addr.IsValid()) {
+                    // If no collisions, consider an address from the "tried" table.
+                    addr = addrman.Select(true);
```

Comment:
I think `true` means pull from the new table, not the tried table, so this line and the one below at line 1946 are backwards.